joomla/joomla-cms
# Added Serializable interface and methods to JTable. Fixes #16185.

Pull Request for Issue #16185 .

### Summary of Changes

Added Serializable interface and methods to JTable. Excluding the DB object from serialization, because under some circumstances the DB object may contain closures; these are unserializable, and thus caused a crash when serializing JTable.

### Testing Instructions

* Install Joomla any of 3.7.0 / 3.7.1 / 3.7.2.
* Install K2 extension.
* Create a K2 category and add some articles to it.
* Create a menu entry for the category view.
* Navigate to the menu entry: It should work.
* Turn on Joomla's debug mode.
* Navigate to the menu entry again: It will crash with `Serialization of 'Closure' is not allowed`.
* Now install the patch.
* Keep debug mode enabled.
* Navigate to the menu entry again: It should now work correctly.

### Expected result

See above: The category view should work correctly in debug mode.

### Actual result

Without the patch, navigating to the category view in debug mode crashes and gives the following error:

> Exception
> Serialization of 'Closure' is not allowed
> .../libraries/joomla/cache/controller/callback.php:184

### Documentation Changes Required

None.
